atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java

```diff
@@ -1103,6 +1103,47 @@ public void testGetRangeOfTimestampsReturnsAllRows() {
     }
 
     @Test
+    public void testGetRangeOfTimestampsOmitsTimestampsLessThanMax() {
+        keyValueService.put(TEST_TABLE,
+                ImmutableMap.of(
+                        Cell.create(row0, column0), value0_t0),
+                TEST_TIMESTAMP);
+
+        keyValueService.put(TEST_TABLE,
+                ImmutableMap.of(
+                        Cell.create(row0, column0), value0_t1),
+                TEST_TIMESTAMP + 10);
+
+        RangeRequest range = RangeRequest.all().withBatchHint(1);
```

Comment:
Could use a batch hint of 2 here, to guard against the weird failure case where we would have returned the later value, but had the earlier value first in the list.